### ISSUE TITLE
[Accessibility]: It is time-consuming to navigate back from the ref panel to the file content pane

### ISSUE THREAD

**Issue by varungandhi-src** [sourcegraph/sourcegraph#36581]
### Audit type

Screen reader navigation

### User journey audit issue

https://github.com/sourcegraph/sourcegraph/issues/33510

### Problem description

After the focus has moved to the ref panel, navigating the focus back to the blob view is quite time-consuming (hold Shift + Tab for long or lots of key presses). This makes using the ref panel tedious.

(This point applies to keyboard navigation too.)

### Expected behavior

I think we should modify the tab hierarchy or the page structure so that one can quickly navigate back from the ref panel to the blob view, perhaps to the specific part where the code intel popover was triggered.

### Assigning labels

- [ ] Please give this issue an estimate by applying a label like `estimate/Xd`, where X is the estimated number of days it will take to complete.
- [X] If this issue is specific to a specific Sourcegraph product, please assign the appropriate team label to this issue.
- [ ] If this issue will require input from designers in order to complete, please assign the label `needs-design`.
- [ ] If you are confident that this issue should be fixed by GitStart, please assign the label `gitstart`.

### Owner

This issue will be fixed by my team, I have assigned a relevant member to it, or I will do so in the near future.

**Comment by umpox** [comment]
@gitstart-sourcegraph Please look at this, I think we can use landmarks to implement this and stay compliant here:

1. Add `role="region"` to `Panel.tsx` and make `aria-label` mandatory
2. Update all usage of `Panel` to have a correct `aria-label`, e.g. the references panel should be "References panel", the sidebar should be "File sidebar".
3. Update `Blob.tsx` to allow `role` and `aria-label` props. In `BlobPage.tsx` use these props to add `role="region"` and `aria-label="File blob"`

See this [PR](https://github.com/sourcegraph/sourcegraph/pull/37577) as an example of how landmarks are used
